src/components/Select.tsx

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { motion, MotionStyle, Transition, Variants } from 'framer-motion'
+import { ChevronDown } from 'lucide-react'
+import { ReactNode, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import useOnOutsideClick from '@/hooks/useOnClickOutside'
+
+interface SelectProps {
+  items: SelectItem[]
+  onItemClick: (value: string) => void
+  selectedItemValue?: string
+  title: string
+  dimensions?: {
+    initialItemHeight?: number
+    expandedItemHeight?: number
+    initialListWidth?: number | string
+    expandedListWidth?: number | string
+  }
+  borderRadius?: number
+  alignText?: 'start' | 'center' | 'end'
+  className?: string
+  style?: MotionStyle
+}
+
+export interface SelectItem {
+  value: string
+  label?: string
+  Component?: ReactNode
+}
+
+const transition: Transition = { type: 'tween', duration: 0.15 }
+
+const Select = ({
+  items,
+  onItemClick,
+  selectedItemValue,
+  title,
+  dimensions = {
+    initialItemHeight: 55,
+    expandedItemHeight: 60,
+    initialListWidth: '98%',
+    expandedListWidth: '100%'
+  },
+  alignText,
+  borderRadius = 12,
+  className,
+  style
+}: SelectProps) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+  const listRef = useRef<HTMLDivElement>(null)
+
+  const { initialItemHeight, expandedItemHeight, initialListWidth, expandedListWidth } = dimensions as Required<
+    typeof dimensions
+  >
+
+  const orderedItems = [
+    ...items.filter(({ value }) => selectedItemValue === value),
```

Comment:
I know this is almost silly optimization, but since you're only looking for 1 item here, this would be more efficient:

```ts
items.find(({ value }) => selectedItemValue === value)
```

Reply:
Done! (I had to cast the type though).